daemon/list.go

```diff
@@ -113,44 +115,70 @@ func (daemon *Daemon) Containers(ctx context.Context, config *containertypes.Lis
 		return nil, err
 	}
 
-	// fastpath to only look at a subset of containers if specific name
+	// shortcut to only look at a subset of containers if specific name
 	// or ID matches were provided by the user--otherwise we potentially
 	// end up querying many more containers than intended
 	containerList, err := daemon.filterByNameIDMatches(view, filter)
 	if err != nil {
 		return nil, err
 	}
+	numContainers := len(containerList)
+	results := make(chan *containertypes.Summary, numContainers)
 
+	// Get the info for each container in the list; this can be slow so we
+	// dispatch a set number of worker goroutines to do the jobs. We choose
+	// log2(numContainers) workers to avoid creating too many goroutines
+	// for large number of containers.
+	numWorkers := log2(numContainers)
```

Comment:
Given that `log2` is literally a 1-line function and we describe here what we do, I think inlining it would make sense (keeping the implementation close to where it's used);

```suggestion
	numWorkers := int(math.Log2(float64(numContainers)))
```

Reply:
I think it's much easier to read `log2(numContainers)` than `int(math.Log2(float64(numContainers)))`, so I would vote to keep it because it reduces mental load when reading the code.

Reply:
Went with your suggestion after all.